contracts/src/components/attack.cairo

```diff
@@ -0,0 +1,63 @@
+use eternum::alias::ID;
+use eternum::utils::coin_toss;
+
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct Attack {
+    #[key]
+    entity_id: ID,
+    chance: u8,
+    last_attack: u64
+}
+
+#[generate_trait]
+impl AttackImpl of AttackTrait{
+    
+    fn get_cost(self: @Attack) -> u8 {
```

Comment:
not sure what you mean by attack cost ? Attacks should be free as long as you are not in the cooloff period

Reply:
get_cost there is to calculate the shekel amount that will be paid for attack when creating a mercenary. E.g if you want a 10% chance of victory during attacks, you pay 1 * 40 shekels. If you want 50 percent chance, you pay 5 * 40 shekels.

Calling the raid system itself doesn't require additional cost